solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/TestMultiSegment.java

```diff
@@ -0,0 +1,113 @@
+      
+      assertU(adoc("id", new Integer(i+11).toString(), "global_categoryId", "201",
+                   "comp_description", "apple apple is a company " + randomString(i%6+3), "global_normHits", "0.1",
+                   "func_displayDescription_posBoosted", "New New York New York is New York is Big New York is Big Apple. "+ randomString(i%6+3),
+                   "func_displayDescription_edge", "New York is Big Apple. "+ randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+12).toString(), "global_categoryId", "201", 
+                   "comp_description", "Big Apple is New York.", "global_normHits", "0.01",
+                   "func_displayDescription_posBoosted", "Big Big Apple Big Apple is Big Apple is New Big Apple is New York. "+ randomString(i%6+3),
+                   "func_displayDescription_edge", "Big Apple is New York. " + randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+13).toString(), "global_categoryId", "201", 
+                   "comp_description", "New some York is Big. "+ randomString(i%6+3),
+                   "func_displayDescription_posBoosted", "New New some New some York New some York is New some York is Big. "+randomString(i%6+3),
+                   "func_displayDescription_edge", "New some York is Big. "+randomString(i%6+3)));
+      
+      assertU(adoc("id", new Integer(i+14).toString(), "global_categoryId", "201",
+          "comp_description", "apple apple is a company " + randomString(i%6+3), "global_normHits", "0.1",
+          "func_displayDescription_posBoosted", "New New York New York is New York is Big New York is Big Apple. "+ randomString(i%6+3),
+          "func_displayDescription_edge", "New York is Big Apple. "+ randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+15).toString(), "global_categoryId", "201", 
+          "comp_description", "Big Apple is New York.", "global_normHits", "0.01",
+          "func_displayDescription_posBoosted", "Big Big Apple Big Apple is Big Apple is New Big Apple is New York. "+ randomString(i%6+3),
+          "func_displayDescription_edge", "Big Apple is New York. " + randomString(i%6+3)));
+      assertU(adoc("id", new Integer(i+16).toString(), "global_categoryId", "401", "ppl_normNewsScore", "0.0", 
+                   "global_normHits", "0.0", "ppl_last_name", "OBAMA " + randomString(i%6+3), "ppl_first_name", "barack h"));
+      assertU(adoc("id", new Integer(i+17).toString(), "global_categoryId", "201", "comp_description", "red delicious apple " + randomString(i%6+3), "global_normHits", "0.1"));
+      assertU(adoc("id", new Integer(i+18).toString(), "global_categoryId", "201", "comp_description", "nyc " + randomString(i%6+3), "global_normHits", "0.11"));
+    }
+    
+    assertU(commit());
+    
+    loadFeatures("comp_features.json");
+  }
+  
+  @AfterClass
+  public static void after() throws Exception {
+    aftertest();
+  }
+  
+  @Test
+  public void testPeopleFirstAndLastMatch() throws Exception {
```

Comment:
I would rename this test, since your purpose here is to verify that you get feature vectors from segments beyond the first